translator/translate/otel/processor/metricstransformprocessor/translator.go

```diff
@@ -31,18 +49,72 @@ func (t *translator) ID() component.ID {
 
 func (t *translator) Translate(conf *confmap.Conf) (component.Config, error) {
 	cfg := t.factory.CreateDefaultConfig().(*metricstransformprocessor.Config)
-	c := confmap.NewFromStringMap(map[string]interface{}{
-		"transforms": map[string]interface{}{
+	transformRules := []map[string]interface{}{
+		{
 			"include":                   "apiserver_request_total",
 			"match_type":                "regexp",
 			"experimental_match_labels": map[string]string{"code": "^5.*"},
 			"action":                    "insert",
 			"new_name":                  "apiserver_request_total_5xx",
 		},
+	}
+
+	if isGpuEnabled(conf) {
+		// appends DCGM metric transform rules for each metric type (container/pod/node) with following format:
+		// {
+		//		"include":  "DCGM_FI_DEV_GPU_UTIL",
+		//		"action":   "insert",
+		//		"new_name": "container_gpu_utilization",
+		//      "operations": [
+		//     		{
+		//				"action":   "add_label",
+		//          	"new_label": "Type",
+		//				"new_value": "ContainerGPU",
+		//			},
```

Comment:
Indentation got messed up here.

Reply:
not sure if it's Github displaying them weirdly. I will re-indent them anyways.